### ISSUE TITLE
Error: Process execution timed out

### ISSUE THREAD

**Issue by mikebronner** [AtomLinter/linter-phpcs#114]
I'm getting this error even in small files:

```
Error: Process execution timed out
    at /Users/mike/.atom/packages/linter-phpcs/node_modules/atom-linter/lib/helpers.js:79:18
```
<img width="460" alt="screen shot 2016-03-03 at 08 17 57" src="https://cloud.githubusercontent.com/assets/1791050/13500711/7de741c8-e118-11e5-8fdd-fe1e99317ce9.png">

Any ideas I might try? Thanks! :)

**Comment by Arcanemagus** [comment]
So the latest version added a 10 second timeout to the execution, if you are hitting this then `phpcs` is screwing up somehow and taking longer than 10 seconds and getting killed. Do you mind sharing a file that is triggering this as well as your configuration that `phpcs` is running on it? Maybe we can figure out why `phpcs` is locking up on it.

**Comment by mikebronner** [comment]
Thanks @Arcanemagus :) I will post here the next time it happens, the configuration for sure, and the file if its not sensitive information (domain logic, etc.).

**Comment by mikebronner** [comment]
OK, just happened again. Here is the file I was in as it happened:

```
<?php

use Illuminate\Foundation\Testing\WithoutMiddleware;
use Illuminate\Foundation\Testing\DatabaseMigrations;
use Illuminate\Foundation\Testing\DatabaseTransactions;

class ExampleTest extends TestCase
{
    /** @test */
    public function it_finds_a_document_by_id()
    {
        
    }
}
```

I was adding the open and closing braces of the test method when it errored. Here are my settings:

<img width="834" alt="screen shot 2016-03-03 at 13 20 20" src="https://cloud.githubusercontent.com/assets/1791050/13509871/bdc12672-e142-11e5-9653-065d2c1e5c22.png">

**Comment by mikebronner** [comment]
So i've had it happen repeatedly, and it seems to occur when I move the cursor between the braces of a method without any content. Will keep an eye out to see if I can get this reproducable.

**Comment by femkhan** [comment]
I'm also experiencing the same issue when i do something inside the parenthesis. 

Sometime this error appears.
`Error parsing PHPCS response
something went wrong attempting to parse the PHPCS output.`

Sometimes it shows same error as shown in above message by @mikebronner

**Comment by Arcanemagus** [comment]
@femkhan if you see that "something went wrong" message can you file a new issue with the file in the exact state you saw that on?

**Comment by Foxtrot** [comment]
Hi,

I'm also experiencing the exact same error as @mikebronner. My settings are below, should it help:

[http://i.imgur.com/TrUZipi.png](url)

**Comment by er3tic** [comment]
Hi,

I'm having the same issue but with linter-clang.

![screen shot 2016-03-18 at 14 56 36](https://cloud.githubusercontent.com/assets/7046950/13879878/12525f52-ed1a-11e5-90e6-d873bf85a084.png)

Here are my settings, but I'm not sure it will help you ;)

![screen shot 2016-03-18 at 15 01 07](https://cloud.githubusercontent.com/assets/7046950/13879928/46949316-ed1a-11e5-8f95-c8c5d8c88198.png)

**Comment by askaper** [comment]
Also having this issue on line 79 of helpers.js except I'm using linter-csslint.

**Comment by Arcanemagus** [comment]
@askaper please file that on `linter-csslint`.

**Comment by neilgee** [comment]
I am having this issue as well
`Error: Process execution timed out
    at /Users/neilg/.atom/packages/linter-phpcs/node_modules/atom-linter/lib/helpers.js:79:18`

Here are my settings:
![screen shot 2016-03-22 at 4 23 13 pm](https://cloud.githubusercontent.com/assets/2056045/13942818/80ceda90-f04a-11e5-9be9-f2cfac53d0a5.png)

Here is an example file - https://github.com/neilgee/slicknav-menu/blob/master/slicknav-mobile-menu.php

**Comment by oknixus** [comment]
me, too.
Error: Process execution timed out
Error: Process execution timed out
    at /Users/KJRS/.atom/packages/linter-phpcs/node_modules/atom-linter/lib/helpers.js:79:18

and my setting:
![image](https://cloud.githubusercontent.com/assets/4592706/14342652/cc8ff8a8-fccd-11e5-8453-a8b984c184fc.png)

**Comment by jyggen** [comment]
Does CodeSniffer seem to hang/take forever when you run it manually as well?

**Comment by miklb** [comment]
I'm getting this error and I'm not even editing php files. The project I have open doesn't have a single php file.

**Comment by Arcanemagus** [comment]
@miklb You need to figure out which linter _is_ running on the files you are editing, I'm assuming since you are commenting here that the path for `atom-linter` contained `linter-phpcs`, but that's just due to Atom having picked that instance of `atom-linter` to load first.

**Comment by oknixus** [comment]
天哪，这问题这么久了，怎么还存在呀！
😭……好像这个也造成Atom变得非常卡！
😭😭😭……

**Comment by Arcanemagus** [comment]
[v1.4.0](https://github.com/AtomLinter/linter-phpcs/releases/tag/v1.4.0) has been published with an option to disable the execution timeout, among a few other new features. Be warned that you may start seeing orphaned processes again if this is enabled!
